### ISSUE TITLE
Unexpected behavior when right clicking item in file browser

### ISSUE THREAD

**Issue by kingwingfly** [zed-industries/zed#21936]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

<img width="1440" alt="Screenshot 2024-12-13 at 04 37 20" src="https://github.com/user-attachments/assets/cd0a5a9f-1ddb-4a16-bfc6-a774eeabbcec" />

As the image shows, I have `agent.py` and `data.pkl`. 

I'm modifying `agent.py`, then I directly right click `data.pkl` and choose delete, it show that zed wants to delete both. However, it is `data.pkl` the only file I wanna delete.

I just deleted my 4-hour work without git backing up _(´ཀ`」 ∠)_...

### Environment

macOS latest beta

zed -v
Zed 0.165.4 – /Applications/Zed.app

### If applicable, attach your Zed.log file to this issue.

unrelated

**Comment by JosephTLyons** [comment]
Hey @kingwingfly. Can I get a bit of extra information here? I notice that both `agent.py` and `data.pk1` are highlighted in the project panel. Did you intentionally select `agent.py` or was that included in some other way? I've noticed that we have a bug where simply hovering the mouse over another project panel entry can include it in project panel actions - is that what happened here?

**Comment by JosephTLyons** [comment]
Actually, I think I misspoke. The issues seems to be that the active file is included in the delete action, when you have focus on the project panel and use the keyboard to delete a file.

**Comment by kingwingfly** [comment]
@JosephTLyons 

Sorry for my poor English, I may be not able to understand `project panel entry` and `panel actions` correctly. 

If I understand correctly:
- `project panel entry` is the file tree on the left
- `panel actions` is the right click menu

Now I'm using Zed 0.167.1, and find that I cannot reproduce the bug anymore. Maybe it's fixed? 

More concretely, assume that we have `a.py` and `b.py`, steps used to lead the bug are as follows: 
1. left click `a.py`.
2. right click `b.py`.
3. choose delete, and both of them are going to be deleted. 

So 
> The issues seems to be that the active file is included in the delete action, when you have focus on the project panel and use the keyboard to delete a file.

is right.

Anyway, it seems fixed already or needing to meet more conditions to reproduce. I'll close this issue.

**Comment by JosephTLyons** [comment]
Yep, I think this specific one was fixed by @0xtimsb, but I do think we can get into a similar issue by selecting and deleting files when using the keyboard vs the mouse.
